micropython/drivers/imu/lsm9ds1/lsm9ds1.py

```diff
@@ -58,129 +59,144 @@
 _OFFSET_REG_X_M = const(0x05)
 _CTRL_REG1_M = const(0x20)
 _OUT_M = const(0x28)
-_SCALE_GYRO = const(((245, 0), (500, 1), (2000, 3)))
-_SCALE_ACCEL = const(((2, 0), (4, 2), (8, 3), (16, 1)))
 
 
 class LSM9DS1:
-    def __init__(self, i2c, address_gyro=0x6B, address_magnet=0x1E):
-        self.i2c = i2c
-        self.address_gyro = address_gyro
+    def __init__(
+        self,
+        bus,
+        address_imu=0x6B,
+        address_magnet=0x1E,
+        gyro_odr=952,
+        gyro_scale=245,
+        accel_odr=952,
+        accel_scale=4,
+        magnet_odr=80,
+        magnet_scale=4,
+    ):
+        """Initalizes Gyro, Accelerometer and Magnetometer.
+        bus: IMU bus
+        address_imu: IMU I2C address.
+        address_magnet: Magnetometer I2C address.
+        gyro_odr: (0, 14.9Hz, 59.5Hz, 119Hz, 238Hz, 476Hz, 952Hz)
+        gyro_scale: (245dps, 500dps, 2000dps )
+        accel_odr: (0, 14.9Hz, 59.5Hz, 119Hz, 238Hz, 476Hz, 952Hz)
+        accel_scale: (+/-2g, +/-4g, +/-8g, +-16g)
+        magnet_odr: (0.625Hz, 1.25Hz, 2.5Hz, 5Hz, 10Hz, 20Hz, 40Hz, 80Hz)
+        magnet_scale: (+/-4, +/-8, +/-12, +/-16)
+        """
+        self.bus = bus
+        self.address_imu = address_imu
         self.address_magnet = address_magnet
-        # check id's of accelerometer/gyro and magnetometer
+
+        ACCEL_SCALE = (2, 16, 4, 8)
+        GYRO_SCALE = (245, 500, 2000)
+        MAGNET_SCALE = (4, 8, 12, 16)
+
+        ODR_IMU = (0, 14.9, 59.5, 119, 238, 476, 952)
+        ODR_MAGNET = (0.625, 1.25, 2.5, 5, 10, 20, 40, 80)
```

Comment:
Is there a reason to have these constants defined within the `__init__` method?  `_SCALE_GYRO` and `_SCALE_ACCEL` were previously constants defined at the top-level of the module, and that makes them efficient (especially in .mpy files).

Can these 5 constant tuples be moved to the top-level scope, and start with underscore to make them true constants?

Reply:
> Is there a reason to have these constants defined within the __init__ method? 

It's because they're only needed temporarily in the `__init__` method, after that they should be collected, this way they don't take any flash/RAM, after init, right ?

Reply:
But the bytecode needs to construct these on the fly, and that takes up flash (or RAM if the code is not frozen) due to larger bytecode.  Storing a const tuple in the .mpy file is probably on par in terms of size of the .mpy file.

Also when floats are involved the code is much bigger that if they are just small ints.

Reply:
Actually I just checked this and it both cases (global or local) it's pretty efficient!  The compiler is optimising these tuples to a true constant, regardless of whether they are defined globally or in the function.

But the global case where they are defined as a `const` is a little more efficient because no local variable (or global) is made.

So, to reduce bytecode size by a little bit, please make them `_SCALE_GYRO = const(...)` etc.